gef-ui/server/controllers/supporting-information/manual-inclusion-questionnaire/index.js

```diff
@@ -0,0 +1,235 @@
+
+const getUploadManualInclusion = async (req, res) => {
+  const {
+    session: { user, userToken },
+    params: { applicationId },
+  } = req;
+
+  const application = await Application.findById(applicationId, user, userToken);
+
+  if (!application) {
+    console.log(`User unauthorised to view application ${applicationId} manual inclusion`);
+    return res.redirect('/dashboard/gef/');
+  }
+
+  return res.render('partials/manual-inclusion-questionnaire.njk', {
+    user,
+    applicationId,
+    files: application.supportingInformation?.manualInclusion,
+  });
+};
+
+const validateFileQuestion = (application, field, errRef) => {
+  const errors = [];
+
+  const files = application?.supportingInformation?.[field] || [];
+
+  if (!files.length) {
+    errors.push({
+      errRef,
+      errMsg: 'Upload Manual inclusion questionnaire',
+    });
+  }
+
+  if (files.length > 20) {
+    errors.push({
+      errRef,
+      errMsg: 'You can only upload up to 20 files at the same time',
+    });
+  }
+
+  return errors;
+};
+
+const postUploadManualInclusion = async (req, res) => {
+  const {
+    body: { delete: fileToDelete, submit },
+    files,
+    params: { applicationId },
+    session: { user, userToken },
+  } = req;
+  const errRef = 'documents';
+
+  let manualInclusionErrors = [];
+  let processedFiles = [];
+
+  let application = await Application.findById(applicationId, user, userToken);
+
+  if (!application) {
+    console.log(`User unauthorised to view application ${applicationId} manual inclusion`);
+    return res.redirect('/dashboard/gef/');
+  }
+
+  // Check if files have been sent via non-JS upload
+  if (files && Array.isArray(files) && files.length) {
+    const validFiles = [];
+    const invalidFiles = [];
+
+    files.forEach((file) => {
+      const [isValid, error] = validateFile(file, MAX_FILE_SIZE);
+
+      if (isValid) {
+        validFiles.push(file);
+      } else {
+        invalidFiles.push({
+          ...file,
+          error,
+        });
+      }
+    });
+
+    const uploadedFiles = validFiles.length ? await uploadAndSaveToDeal(
+      validFiles,
+      FIELD_NAME,
+      application,
+      userToken,
+      MAX_FILE_SIZE,
+    ) : [];
+
+    processedFiles = [
+      ...invalidFiles,
+      ...uploadedFiles,
+    ];
+
+    manualInclusionErrors = processedFiles.reduce((fileErrors, file) => {
+      if (file.error) {
+        fileErrors.push({
+          errRef: file.filename || file.originalname,
+          errMsg: file.error,
+        });
+      }
+
+      return fileErrors;
+    }, []);
+  }
+
+  // Need to re-get the application here if any file changes
+  if (processedFiles.length) application = await Application.findById(applicationId, user, userToken);
+
+  if (fileToDelete) {
+    try {
+      await removeFileFromDeal(fileToDelete, FIELD_NAME, application, userToken);
+    } catch (err) {
+      const errMsg = `Error deleting file ${fileToDelete}: ${err.message}`;
+      console.log(errMsg);
```

Comment:
would be better to do console.error for debugging in sentry, not a blocker for this PR

Reply:
good call